### ISSUE TITLE
firefox webRTC: ICE failed

### ISSUE THREAD

**Issue by amirhosseinAshnavar** [OpenVidu/openvidu#536]
I am using openvidu-browser@2.15.0 on angular and 
openvidu-server@2.15.0 (on vm using ubuntu18.04) it is working well on chrome but on firefox we face an error which i will attach its screenshot to this post. i have looked up issues related to this problem on github but they didn't solve my problem . i think there is something wrong with openvide server (i deployed openvidu server using open vidu docs and opened needed ports ) 
![firefoxBug](https://user-images.githubusercontent.com/61137724/92438933-1618e800-f1bf-11ea-9abe-68b1321a5bed.jpg)

**Comment by pabloFuente** [comment]
Are you able to establish a connection in https://demos.openvidu.io with that Firefox browser?

**Comment by amirhosseinAshnavar** [comment]
> Are you able to establish a connection in https://demos.openvidu.io with that Firefox browser?

Yes , and this is why I told I think its a server problem .

**Comment by pabloFuente** [comment]
If you are able to do so, then your deployment must have some kind of network restriction that is making the ICE negotiation fail.
I know you have said this: `i deployed openvidu server using open vidu docs and opened needed ports`, but ICE failures are 99.99% of the times caused by the networking of the deployment not being fine.  For OpenVidu CE, double check that this requirements are properly met, including the port ranges for KMS and TURN:

![image](https://user-images.githubusercontent.com/15049534/92489377-006fe680-f1f0-11ea-9cff-d0eeb9f8c261.png)

**Comment by amirhosseinAshnavar** [comment]
> If you are able to do so, then your deployment must have some kind of network restriction that is making the ICE negotiation fail.
> I know you have said this: `i deployed openvidu server using open vidu docs and opened needed ports`, but ICE failures are 99.99% of the times caused by the networking of the deployment not being fine. For OpenVidu CE, double check that this requirements are properly met, including the port ranges for KMS and TURN:
> 
> ![image](https://user-images.githubusercontent.com/15049534/92489377-006fe680-f1f0-11ea-9cff-d0eeb9f8c261.png)

I have just checked the server , my openvidu server is deployed by docker on a vm and both vm and its host machine needed ports are open as you can see in attached file . 
![portsSS](https://user-images.githubusercontent.com/61137724/92563923-97d74700-f28d-11ea-8a36-bcd2febce80a.jpg)

**Comment by amirhosseinAshnavar** [comment]
> If you are able to do so, then your deployment must have some kind of network restriction that is making the ICE negotiation fail.
> I know you have said this: `i deployed openvidu server using open vidu docs and opened needed ports`, but ICE failures are 99.99% of the times caused by the networking of the deployment not being fine. For OpenVidu CE, double check that this requirements are properly met, including the port ranges for KMS and TURN:
> 
> ![image](https://user-images.githubusercontent.com/15049534/92489377-006fe680-f1f0-11ea-9cff-d0eeb9f8c261.png)

Any solutions ??

**Comment by aloisogomes** [comment]
hello @amirhosseinAshnavar i have same problem  and everything works fine. 

if you have native (not dockerized) COTURN and KMS installed on your server you must follow these steps:

1) Stop your docker of openvidu server: `./openvidu stop`
2) Stop Kurento service: `service kurento-media-server stop`
3) Stop Coturn service: `systemctl stop coturn`
4) Start docker of openvidu server: `./openvidu start`

Best regards

**Comment by sergencansiz** [comment]
Hello,

I am also having same ICE Failed error on Firefox browsers and some times in Chrome too. I deployed Openvidu 2.15 on Azure Ubuntu 18.04 by following this instruction: https://docs.openvidu.io/en/2.15.0/deployment/deploying-on-premises/. I am getting this error in openvidu-call demo which comes with "Deploy > On Premises 2.15" as ready to use. I thought  Cotrun might cause this error but  I have checked Coturn on Trickle ICE and it works okay with given username and pass to the client by OpenVidu server. All UDP and TCP ports are available as it is indicated in the instructions.  There is not any Warning or Error in OpenVidu logs.  I also developed an basic application by using openvidu-broswer on react it also gives same error on Firefox and some times on other browser.   Also checked https://demos.openvidu.io/openvidu-call/#/ demo with same browsers it works fine no error for firefox.

This problem occurs when OpenVidu is deployed on cloud. I didn't faced this problem in development setup by using openvidu-server-kms docker image. I tested with several machines in the same LAN and it work fine.  Is there any other way to deploy OpenVidu?

Thank you

**Comment by aloisogomes** [comment]
> 
> 
> Hello,
> 
> I am also having same ICE Failed error on Firefox browsers and some times in Chrome too. I deployed Openvidu 2.15 on Azure Ubuntu 18.04 by following this instruction: https://docs.openvidu.io/en/2.15.0/deployment/deploying-on-premises/. I am getting this error in openvidu-call demo which comes with "Deploy > On Premises 2.15" as ready to use. I thought Cotrun might cause this error but I have checked Coturn on Trickle ICE and it works okay with given username and pass to the client by OpenVidu server. All UDP and TCP ports are available as it is indicated in the instructions. There is not any Warning or Error in OpenVidu logs. I also developed a basic application by using openvidu-broswer on react and it also gives same error on Firefox and some times on other browser. Also checked https://demos.openvidu.io/openvidu-call/#/ demo with same browsers, it works fine no error for firefox.
> 
> This problem occurs when OpenVidu is deployed on cloud. I didn't faced this problem in development setup by using openvidu-server-kms docker image. I tested with several machines in the same LAN and it worked fine. Is there any other way to deploy OpenVidu?
> 
> Thank you

HI, i'm faced same issue. Do you stopped KMS and Coturn  native services before start Openvidu's Docker?

**Comment by sergencansiz** [comment]
Hi,
I found the problem. We were running OpenVidu server on Azure virtual machine.  Machine was open to only certain IP addresses for security reasons. After we disabled and open the port 3478 TCP+UDP (coturn) to all IPs problem is solved. But there is an interesting thing here.  It was okay for chrome when we access the application from defined IPs but not okay for Firefox. I guess when Firefox request SDP for WebRTC over the ICEServers, it pass from middle process which calls TURN server from different IP. I didn't really understand that part . Maybe someone who has experience on it can answer that issue.
